doc/bmp/bmp.md

```diff
@@ -0,0 +1,373 @@
+   - Description: Enable the BGP neighbor table.  
+   - Result: reset FRR connection and populate BGP_NEIGHBOR_TABLE table
+   
+2. Command: `config bmp disable bgp-neighbor-table`  
+   - Description: Disable the BGP neighbor Capability table.  
+   - Result: erase BGP_NEIGHBOR_TABLE table and stop table population
+   
+3. Command: `config bmp enable bgp-rib-out-table`  
+   - Description: Enable the BGP rib advertised table to all neighbors.  
+   - Result: reset FRR connection and populate BGP_RIB_OUT_TABLE table
+
+4. Command: `config bmp disable bgp-rib-out-table`  
+   - Description: Disable the BGP rib advertised table to all neighbors.  
+   - Result: erase BGP_RIB_OUT_TABLE table and stop table population
+   
+5. Command: `config bmp enable bgp-rib-in-table`  
+   - Description: Enable the BGP rib received table from all neighbors.  
+   - Result: reset FRR connection and populate BGP_RIB_IN_TABLE table
+
+6. Command: `config bmp disable bgp-rib-in-table`  
+   - Description: Disable the BGP rib received table from all neighbors.  
+   - Result: erase BGP_RIB_IN_TABLE table and stop table population
+ 
+```
+
+bmp will support below show CLIs to query specific table in runtime:
+
+```
+1. Command: `show bmp bgp-neighbor-table`  
+   - Description: Show BGP neighbor table for all neighbors.  
+   - Result: Query BGP_NEIGHBOR_TABLE table and show result in human readable format.
+   
+2. Command: `show bmp bgp-rib-out-table`  
+   - Description: Show BGP routes advertised table for all neighbors.  
+   - Result: Query BGP_RIB_OUT_TABLE table and show all session data.
+
+3. Command: `show bmp bgp-rib-in-table`  
+   - Description: Show BGP routes received table for all neighbors.  
+   - Result: Query BGP_RIB_IN_TABLE table and show all session data
+
+4. Command: `show bmp table status`  
+   - Description: Show all table status like enable or disable so that user could operate correctly.  
+   - Result: Query and show config db status for all table enablement
+
+5. Command: `show bmp out prefix count`
+   - Description: Show BGP prefixes advertised total count 
+   - Result: Query BGP_RIB_OUT_TABLE table and show calculated result in human readable format.
+
+6. Command: `show bmp in prefix count`
+   - Description: Show BGP prefixes received total count.  
+   - Result: Query BGP_RIB_IN_TABLE table and show calculated result in human readable format.
+
+```
+
+
+# 4 Resource usage and Test plan
+
+### Resource usage
+Since BMP is a sidecar daemon for improve the SONiC debuggability and monitor efficiency, we don't expect it take too much CPU usage, and prefer it could be able to work under low priority but without break. Now FRR to openbmp is working under push model, if limit cpu of bmp into 5%, it just operate slow, but without missing update, when we flap the bgp several times, all the flap updates will pushed to bmp finally. Thus we should be able to restrist CPU of BMP to lower priority without
```

Comment:
There was an exercise done optimize memory usage in DASH WG. Since we may store lot of RIB information into the DB, may be it's ideal to use protobuf format rather than text format for DB entries (refer the below link for more information).
https://github.com/sonic-net/DASH/blob/main/documentation/general/data/AppDBMemoryEstimation.xlsx

Reply:
Yes, will make some comparison for this, using pb should be feasible just the data will not be readable if we use redis-cli

Reply:
yes, most database contents are in text format, we may need to check the possibility of memory optimization as we may push lot of RIB information into the DB.